### PR TITLE
docs - start adding FDW using and developing content

### PR DESCRIPTION
**Pull request by lisakowen**
**this PR** - provide some basic FDW info, continue to fill out FDW doc content before GA:
- tried to minimize the use of the term "external" when discussing foreign data
- add foreign data/tables introduction, placed in parallel with external table content
- add a "Writing a Foreign Data Wrapper" topic - some introductory usage/build information, greenplum database considerations

**future topics**:
- more development and greenplum considerations
- table and column-level options
- fdw development example

**questions**:
- the greenplum 6 beta does not appear to include any foreign data wrappers.  do we plan to include at least file_fdw at some point before or at 6.0 GA?
- is insert/update of a foreign table supported in the greenplum 6.0 beta?
- added a "known issues and limitations" section.  are the items correct?  are there others we should add?
- mpp_execute - are the options (master/all/any) referring to hosts or segment instances or ??

**doc review site links**:
- external data landing page - http://docs-lisa-fdw.cfapps.io/6-0/admin_guide/external/g-working-with-file-based-ext-tables.html
- new intro page for foreign tables - http://docs-lisa-fdw.cfapps.io/6-0/admin_guide/external/g-foreign.html#topic1
- new page describing writing an FDW - http://docs-lisa-fdw.cfapps.io/6-0/admin_guide/external/g-devel-fdw.html#topic1

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/external/g-working-with-file-based-ext-tables.xml [hunk 1]
```diff
@@ -2,15 +2,30 @@
...
+      table. For example, you can <codeph>SELECT</codeph> (readable external
+      table), <codeph>INSERT</codeph> (writable external table), and join
+      external tables with other Greenplum tables. External tables are most
+      often used to load and unload database data. Refer to
+      <xref href="g-external-tables.xml#topic3"/> for more information about
+      using external tables to access external data.</p>
+    <p><xref href="pxf-overview.xml" type="topic">Accessing External Data with PXF</xref>
+      describes using PXF and external tables to access external data sources.</p>
+    <p>A foreign table is a different kind of Greenplum Database table backed with
+      data that resides outside of the database. You can both read from and write
```
Comment (frankgh) [inline gpdb-doc/dita/admin_guide/external/g-working-with-file-based-ext-tables.xml]: `write` only when the FDW supports it. Not sure if we need to be that specific here though

File: gpdb-doc/dita/admin_guide/external/g-devel-fdw.xml [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+	ExplainForeignModify_function ExplainForeignModify;
+
+	/* Support functions for ANALYZE */
+	AnalyzeForeignTable_function AnalyzeForeignTable;
+} FdwRoutine;
+</codeblock>
+      <p>You must implement the scan-related functions in your foreign-data
+        wrapper; implementing the other callback functions is optional.</p>
+      <p>Scan-related callback functions include:</p>
+      <table id="in201681">
```
Comment (frankgh) [inline gpdb-doc/dita/admin_guide/external/g-devel-fdw.xml]: should we add callbacks for routines for updating foreign tables? (i.e `ExecForeignInsert`)

**Comment by lisakowen** [comment]
thank you for reviewing, @frankgh !  we plan to continue to add to the FDW docs up until GA.  will address updating foreign tables in a future PR.

**Comment by lisakowen** [comment]
planning to merge, will address any comments in a follow-up PR.  thanks!

**Pull request merged**